### ISSUE TITLE
Suggestion: Pulverize to explore twinkly powder effects

### ISSUE THREAD

**Issue by Phate4569** [soolar/sl_ascend#239]
While not essential, if a character can pulverize useless items they should. Some path/star combinations have very good twinkly powder effects that are worth spamming if they don't conflict w/ goals. and they are very easy to get in plentiful amounts.

Currently in SC/Opossum I have "Broberry Brotality", which has assisted the script in keeping me from dying (mostly)

**Comment by taltamir** [comment]
To clarify, you mean for it to be done specifically in 2RCS path? because in all other paths it just gives DR3 which is quite useless. But from you saying SC/Opossum i think you mean the 2RCS combo seed of sealclubber with opossum sign?

**Comment by Phate4569** [comment]
Correct, sorry for not being clear. In 2CRS the results of pulverizing can be much more rewarding.

**Comment by taltamir** [comment]
Alright, thank you for clarifying. Well, it seems you have a good point there. Pulverizing in general in 2CRS can be quite good. Not just into twinkly wads. Powders and nuggets too (Actually they should have preference since they are cheaper and worth a lot less outside of 2CRS). and also of various elements (hot, cold, stench, spooky, sleaze). 

However, it is also an utterly massive project to implement due to the sheer number of combinations involved. Anyways, looking at 
http://plasticlobster.com/summer/items.php
I can see many uses for twinkly powder

Notable combinations are:
twinkly powder AT/platypus +33% item for 51 turns
twinkly powder AT/wallaby +20% meat (nuns, meat wall)
twinkly powder AT/wombat +varied meat (nuns, meat wall)
twinkly powder PM/marmot +100% all stats (tower tests)
twinkly powder PM/platypus +200% mox and spell dmg (tower test)
twinkly powder PM/packrat +50% item
twinkly powder PM/vole +3 hot res (kitchen)
twinkly powder PM/wallaby +50 init (for modern zmobies in crypt)
twinkly powder PM/wombat +25 init (for modern zmobies in crypt)
twinkly powder SR/blender +100 init (for modern zmobies in crypt)
twinkly powder SR/platypus +10 hot dmg, +10 hot spell dmg (tower hot test)
... yea ok that is only half of them and its already exhausting.

Is it possible to maybe handle this procedurally?

**Comment by jaspercb** [comment]
One possible procedural way to handle this:

Have a function, `tcrsAcquireBuff(string toMaximize, float meatPerUnit)`, that takes
1) an effect type: "hot damage", "moxie", whatever
2) a cutoff for how much meat it's willing to pay per unit of the above

We can then use the record-return-valued version of https://wiki.kolmafia.us/index.php?title=Maximize to look through available consumables, perhaps filter them to a list of items we know are renewable or that we currently have >5 of (e.g. certain wads, any potions that are purchasable in a shop), filter them again by meat-cost-to-acquire, and use 'em.

**Comment by taltamir** [comment]
doesn't that require first pulverizing in order to make such consumable available in the first place?

**Comment by Phate4569** [comment]
Not really, all of the spaded data is available in the TCRS_<CLASS>_<SIGN>.txt files. You can pull it from there or use the effect_modifier() function.

https://wiki.kolmafia.us/index.php?title=Effect_modifier

**Comment by Phate4569** [comment]
Not really, all of the spaded data is available in the TCRS_[CLASS]_[SIGN].txt files. You can pull it from there or use the effect_modifier() function.

https://wiki.kolmafia.us/index.php?title=Effect_modifier

**Comment by Phate4569** [comment]
Update: I watched it closely today and it seems to be using Powder/nuggets properly (not where I would have, but still in a logical manner).

Possibly update goal to pulverize all non-essential items? For example:
- Pulverize all but 1 of each type of Head, back, torso, pants, offhand items (instead of having 2-3 of the same shirt we don't need)
- Pulverize all but 3 of each accessory, 2 of each 1 hand weapon, 1 of each 2 hand weapon.

this way we generate a lot of potions the script can use.

**Comment by taltamir** [comment]
well, normally those items get sold for meat to finance the run. maybe only pulverize if meat is over 2000 and the results of the pulverize are not redundant. (aka, already have over 5 of the powder/nuggets/whatever it would generate)

**Comment by Phate4569** [comment]
I've rarely let ot get that far. With my perms (+70% meat maybe), equipment, and potions in 2CRS I'm rinning a near permanent 200-400% Meat gain. So except for EARLY run I have meat to spare.

2CRS is just such an odd path, because of the way effects and enhancements are seeded they are much more beneficial.

I dunno. It isn't required, but the resultz of Pulverizing are quite intense.
